### PR TITLE
feat(adapters/surrealdb): update surrealdb.js to enable http connection strategy

### PR DESCRIPTION
**Pull request by martinschaer**
## ☕️ Reasoning

A bug was fixed on surrealdb.js 0.9.1 which enables the HTTP strategy: https://github.com/surrealdb/surrealdb.js/pull/171

This PR includes unit tests for both strategies (RPC and HTTP) and docs.

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

No open issues. Continuation of https://github.com/nextauthjs/next-auth/pull/6251

### CODE REVIEW

File: packages/adapter-surrealdb/src/index.ts [hunk 1]
```diff
@@ -83,6 +105,87 @@ export const sessionToDoc = (
...
+ * export default clientPromise
+ * ```
+ *
+ * #### Option 2/2 – Using HTTP:
+ *
+ * Usefull in serverlees environments like Vercel.
+ *
+ * ```js
+ * import { ExperimentalSurrealHTTP } from "surrealdb.js"
+ * import fetch from "node-fetch"
```
Comment (ThangHuuVu) [inline packages/adapter-surrealdb/src/index.ts:149]:
```suggestion
 * import { ExperimentalSurrealHTTP } from "surrealdb.js"
```

File: packages/adapter-surrealdb/tests/index.test.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
-import Surreal from "surrealdb.js"
+import Surreal, { ExperimentalSurrealHTTP } from "surrealdb.js"
 import { runBasicTests } from "@auth/adapter-test"
+import fetch from "node-fetch"
```
Comment (ThangHuuVu) [inline packages/adapter-surrealdb/tests/index.test.ts:3]:
```suggestion
import { runBasicTests } from "@auth/adapter-test"
```

**Comment by ThangHuuVu** [comment]
seems like fetch is missing for some reason - but it should be available in Node since v18 🤔 strange, will check more when I have time

**Comment by martinschaer** [comment]
> seems like fetch is missing for some reason - but it should be available in Node since v18 🤔 strange, will check more when I have time

@ThangHuuVu turned out fetch is missing just on the global scope of jest. 88e2ab3 adds fetch to jest's globals

**Review by ThangHuuVu — APPROVED** [review]

**Comment by ThangHuuVu** [comment]
thanks! merging, some other tests are failing but we will take another look later.

**Pull request merged**